tests/cpp/multidevice.cpp

```diff
@@ -33,6 +42,34 @@ MultiDeviceTest::~MultiDeviceTest() {
   }
 }
 
+void MultiDeviceTest::waitForDebugger() {
+  char* rank_to_debug_str = getNvFuserEnv("MPI_DEBUG_RANK");
+  if (rank_to_debug_str == nullptr) {
+    return;
+  }
+  const DeviceIdxType rank_to_debug = std::stol(rank_to_debug_str);
+  NVF_CHECK(
+      rank_to_debug >= 0 && rank_to_debug < communicator->size(),
+      "rank_to_debug=",
+      rank_to_debug,
+      " must be in the range of [0,",
+      communicator->size(),
+      ").");
+
+  if (communicator->deviceId() == rank_to_debug) {
+    volatile bool waiting = true;
+    auto pid = getpid();
+    std::cerr << "Process " << pid
+              << " is waiting for the debugger. To continue debugging, "
+              << "start gdb, `attach " << pid
+              << "`, `set var waiting=false`, and `fini`." << std::endl;
+    while (waiting) { // Please change `waiting` in the debugger.
+    }
+    std::cerr << "Process " << getpid() << " finished waiting." << std::endl;
+  }
+  communicator->barrier();
+}
+
 void MultiDeviceTest::SetUp() {
```

Comment:
I probably missed it from a previous PR, but why do we now have a class constructor and a `SetUp` function? I think it would be more readable to keep everything in the same place.

Reply:
gtest [advocates](https://google.github.io/googletest/faq.html#CtorVsSetUp) for constructors for good reasons. It doesn't explicitly speak about both-constructor-and-setup (good question!), but AFAIU the pros and cons mentioned there do still apply. So I changed SetUp do GTEST_SKIP only because that cannot be done in a constructor, and moved the rest to the constructor.

Reply:
> gtest [advocates](https://google.github.io/googletest/faq.html#CtorVsSetUp) for constructors for good reasons. It doesn't explicitly speak about both-constructor-and-setup (good question!), but AFAIU the pros and cons mentioned there do still apply.

IIUC, the two reasons why they advocate for constructors/destructors over SetUp/TearDown  are 1) using `const` and 2) not having to call parent's class SetUp/TearDown.
But in the present case, none of these motivations applies, right?

Reply:
> using const

[These fields](https://github.com/NVIDIA/Fuser/blob/4e8356915fc6ffa823cf97fed4249cc00f317fdf/tests/cpp/multidevice.h#L36-L46) in MultiDeviceTest should have been constant. 

> not having to call parent's class SetUp/TearDown.

You are right in this particular instance. It's still a good practice to minimize SetUp/TearDown because people might not understand this intricacy before copying the code and tend to overuse SetUp. (It's especially tempting to overuse SetUp because people tend to incorrectly think SetUp is per test while the constructor is per test fixture)